pre_commit/clientlib.py

```diff
@@ -225,7 +246,7 @@ def warn_unknown_keys_repo(
     *(
         cfgv.OptionalNoDefault(item.key, item.check_fn)
         for item in MANIFEST_HOOK_DICT.items
-        if item.key != 'id'
+        if item.key not in ['id', 'files', 'exclude']
```

Comment:
these should still get applied -- there's no harm in checking a key more than once

just a tip though for python, you'd be better to write `if item.key not in {'id', 'files', 'exclude'}` (as a set) than as a list